### DISCUSSION TITLE
how do we restart the kafka cluster using the strimzipodsets

### DISCUSSION THREAD

**Discussion by hari819** [strimzi/strimzi-kafka-operator#7600] [Q&A] [answered]
@scholzj  ,

we have migrated recently to version 0.31.1 , with kafka version as 3.2.3 , previous versions had these statefulsets for restarting kafka and zookeeper in sequence to restart the kafka cluster , but now if i try to scaledown the same way using strimzipodsets , it is throwing error , but i can get data in command , 

kubectl get sps -n kafka

so please could you help me scaling down and scaling up strimzipodsets , otherwise everytime i had to restart the kafka cluster , i had to remove the cluster definition and redeploy it , which is scary in production , please could you help

Thanks,

**Comment by scholzj** [comment]
If you want to scale the cluster up or down, you always have to do it in the Kafka custom resource. It was like that already before with StatefulSets and it remains this way with StrimziPodSets.

**Reply by hari819** [reply]
but i was able to scaledown , operator deployment , kafka and zookeeper statefulset previously .

I am unable to run the scale command on strimzipodsets to use , replica=0 , is it possible ?

**Reply by scholzj** [reply, marked as answer]
You cannot scale Kafka to 0. It is a stateful application and the data have to live somewhere. 

If you wanna stop it, then call it that way because it is not scaling and its confusing. And if you want to stop it, you can just stop the operator and delete the StrimziPodSet resource and have the operator recreate it when you want to continue with the cluster.

**Reply by hari819** [reply]
thankyou @scholzj  , i have adjusted my script in the sequence you suggested , 

thankyou

**Reply by ervikrant06** [reply]
Hello Jakub, 

A relevant question related to strimzipodsets. Watched this [video](https://www.youtube.com/watch?v=iSwrn1Gumx4)

I couldn't figure out how we have different resource spec for kafka brokers while using strimzipodsets with 0.32 (which have them by default). Hypothetical example: Like with total replica of 4 having two 2 brokers with high mem/cpu reqt. and 2 brokers with bit low mem/cpu reqt, is't possible to do?
 Also if I want to scale down the particular strimzipodset replica how can we do it. Earlier sts was in decreasing order, strimzipodset looks to be more flexbility in that context.

**Reply by scholzj** [reply]
That is not supported currently - that is a plan for the future.
